### DISCUSSION TITLE
github.dev: one-click rebase (and squash)

### DISCUSSION THREAD

**Discussion by fulldecent** [community/community#70738] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

# Pain point

The number one actionable feedback I get on my pull requests is "please rebase on master" and sometimes "squash and rebase".

This causes the continuous integration to re-run. And also it shows that you are still alive, which increases the chances of getting your PR merged. Especially if you wont on a lot of open source projects where nobody knows who is likely to be online this month.

(Do you have a lot of PRs open? Check here: https://github.com/pulls)

Here is the current best process to do this:

```sh
cd Desktop
mkdir tmp && cd tmp
git clone git@github.com:fulldecent/jekyll.git
cd jekyll
git checkout patch-1
git remote add upstream https://github.com/jekyll/jekyll.git
git fetch upstream
git rebase upstream/master
git push origin patch-1 --force
```

This also requires substituting in your repo name, your branch name, upstream repo name, upstream branch name where appropriate.

Rarely, but sometimes (10%, N=100), this process requires a little finessing because of a merge conflict. Such a scenario can be handled easily in github.dev.

# Feature request

On the pull request page, you should be able to press `.` to load github.dev (this part works) and then something in this interface should allow to rebase on the upstream branch.

Currently I don't see that is possible in github.dev. But requesting this to be possible and highly visible. 

## Source

I was recommended to leave this feedback here by https://docs.github.com/en/codespaces/the-githubdev-web-based-editor

**Comment by patricktokeeffe** [comment]
Indeed, would be a fantastic addition. No clue when or if it will ever be released though...

See [Git: Support interactive rebase #31153](https://github.com/microsoft/vscode/issues/31153#top)
